waku/waku_sync/protocol.nim

```diff
@@ -66,14 +66,14 @@ proc request(self: WakuSync, conn: Connection): Future[Result[seq[WakuMessageHas
   let negentropy = Negentropy.new(self.storage, DefaultFrameSize)
 
   let payload =  negentropy.initiate().valueOr:
-    free(negentropy)
-
+    negentropy.delete()
```

Comment:
We could have something like:
```
defer:
  negentropy.delete()
```
That way all possible return cases are covered.